### PR TITLE
Update CHANGELOG.md with v0.10.0 and bump version to 0.10.1

### PR DESCRIPTION
**Pull request by amartinezfayo**
Update CHANGELOG.md for v0.10.0.

### CODE REVIEW

**Comment by evan2645** [comment]
This is looking great, thank you Agustin! I left a few small comments on the existing lines. I also dug a bunch of stuff up from history that wasn't reflected here but I thought could/should be? Here are the ones I found

> - Fixed a bug which resulted in incorrect SHA for certain types of workloads (#1405)
> - OIDC Discovery Provider now supports listening on a Unix Domain Socket (#1408)
> - Fixed a bug that could lead to agent eviction if a crash occurred during agent SVID rotation (#1399)
> - The `upstream_bundle` configurable now defaults to true, and is marked as deprecated (#1404)
> - OIDC Discovery Provider and the Kubernetes Workload Registrar release binaries are now available via the `spire-extras` tarball (#1424)
> - Introduced new plugin type UpstreamAuthority, which supports both X509-SVID and JWT-SVID as well as the ability to push upstream changes into SPIRE Server (#1388, #1394, #1406, #1455)
> - AWS PCA, AWS Secrets, Disk and SPIRE UpstreamCA plugins have been ported to the UpstreamAuthority type (#1411, #1409, #1410, #1415)
> - Introduced a new RPC `PushJWTKeyUpstream` in the Node API for publishing JWT-SVID signing keys from downstream servers (#1416)
> - Introduced a new RPC `FetchBundle` in the Node API for fetching an up-to-date bundle (#1458)
> - AWS PCA UpstreamAuthority plugin endpoint now configurable (#1498)

At least a few of these are covered by the much larger "JWT-SVID now works in nested topology" update, but I think they are still important/significant enough to call out independently. New plugin interface type, new RPCs that could be of interest security-wise, etc, all seem relavent to me.

**Comment by evan2645** [comment]
Another thing I just thought of that is probably worth noting here too:
> - The UpstreamCA plugin interface is now marked as deprecated in favor of the UpstreamAuthority plugin

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## [0.10.0] - 2020-04-15
+- Added support for JWT-SVID in nested SPIRE topologies (#1388, #1394, #1396, #1406, #1409, #1410, #1411, #1415, #1416, #1417, #1423, #1440, #1455, #1458, #1469, #1476)
+- Added targeted caching to reduce database load (#1439)
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply CHANGELOG.md:5]: `Reduced database load under certain configurations` looks good to me!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+- Agent now proactively rotates workload SVIDs in response to registration updates (#1441, #1477)
+- Small telemetry improvements (#1445)
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply CHANGELOG.md:7]: I would keep it in the changelog with your suggested change, for the same reason that you point. I think it's worth noting that a telemetry counter was removed.

**Comment by amartinezfayo** [comment]
> At least a few of these are covered by the much larger "JWT-SVID now works in nested topology" update, but I think they are still important/significant enough to call out independently. New plugin interface type, new RPCs that could be of interest security-wise, etc, all seem relavent to me.

I was not sure about separately include items that are already included by a parent item. I see now more clearly that a lot of things would be hidden otherwise. Thanks!
I'm also realizing that I left some items out because I was looking at the wrong starting point for the release.

**Comment by azdagron** [comment]
Maybe one day we'll figure out a convenient contribution process without too much overhead to facilitate a linear history with one commit per PR :trollface:

**Comment by amartinezfayo** [comment]
> Compiling the list can be trickier than it seems and I usually use a few different methods and cross reference.

Thanks for the tips @evan2645!

**Pull request merged**